modules/ROOT/pages/policy-openid-connect.adoc

```diff
@@ -8,98 +8,182 @@ endif::[]
 
-[listing]
-anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>
+Alternatively, you can also specify this parameter in the `wrapper.conf` file.
+ 
+To control the number of tokens that can be cached simultaneously, specify the following property when starting the Mule runtime engine:
+ 
+`anypoint.platform.openid_connect_cache_max_size=<amount of tokens to be cached; default value is 10.000>`
 
-Setting the previous property to true makes the policy use the runtime proxy settings if specified. For example:
+Alternatively, you can also specify this parameter in the `wrapper.conf` file.
 
-----
-anypoint.platform.proxy_host=localhost
-anypoint.platform.proxy_port=8080
-----
+[[token-validation-endpoint-authentication]]
+=== Token Validation Endpoint Authentication
 
-== Leveraging OAuth provider information
+To complete validating the provided token, the policy sends a request to the validation endpoint. The policy then obtains the credentials from the client provider configuration (Token Introspection Client section) in Access management. These credentials are sent in the request body, as specified in the https://tools.ietf.org/html/rfc6749#section-2.3[Client Authentication section of the OAuth 2.0 Authorization Framework specification]. 
 
-When a token is successfully validated by the OAuth provider, the provider sends back to the policy who requested the token validation, some extra information, including fields that are OpenID Connect side configurable.
+Other authentication methods for the token validation endpoint are currently not supported.
 
-Token validation response example from the authorization server:
+== Configuring the Expose Headers Option
+
+If you configured the `Expose Headers` option in the policy, the application (for example, out-of-the-box proxies) uses the HTTP requester to redirect the properties to the backend as headers. 
+
+For each field, the original request to the protected resource is enriched with the HTTP headers using the format: X-AGW- + <key>=<value>. For the example response from the previous section, the following headers will be added:
 
 [source,text,linenums]
-----
-{
-    "uid":"john.doe",
-    "mail":"john.doe@example.com",
-    "scope":["uid","mail","cn","givenName"],
-    "grant_type":"password",
-    "cn":"John Doe",
-    "realm":"/",
-    "token_type":"Bearer",
-    "expires_in":3600,
-    "givenName":"John",
-    "access_token":"fa017a0e-1bd5-214c-b19d-03efe9f9847e"
-}
-----
+---
+X-AGW-uid=john.doe
+X-AGW-mail=john.doe@example.com
+X-AGW-token_type=Bearer
+X-AGW-expires_in=3600
+---
+
+For further processing within the flow, the original unparsed response from the validation endpoint is made available. The information is stored in an authentication object. 
+
+For example, suppose a `"mail`" field is returned to the policy, you can access the value of the field by using the following DataWeave 2.0 expression:
```

Comment:
"suppose...you can access" is ungrammatical:
```suggestion
For example, if a `"mail`" field is returned to the policy, you can access the value of the field by using the following DataWeave 2.0 expression:
```

Reply:
Done! Thanks!